src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs

```diff
@@ -46,12 +45,27 @@ public static class Keywords
         private PollingCounter? _lohSizeCounter;
         private PollingCounter? _pohSizeCounter;
         private PollingCounter? _assemblyCounter;
-#endif // !NATIVEAOT
-
         private PollingCounter? _ilBytesJittedCounter;
         private PollingCounter? _methodsJittedCounter;
         private IncrementingPollingCounter? _jitTimeCounter;
 
+#if NATIVEAOT
+        /// <summary>
+        /// If EventSource feature is enabled, RuntimeEventSource needs to be initialized
+        /// In CoreCLR, this is done via StartupHookProvider.CoreCLR.cs
+        /// </summary>
+#pragma warning disable CA2255
+        [ModuleInitializer]
+        internal static void NativeAOTtartupHook()
```

Comment:
```suggestion
        internal static void NativeAOTStartupHook()
```

Reply:
Or better yet - just put the `ModuleInitializer` on the `Initialize` method directly. The entire body of it is under `IsSupported` already and we're not getting anything out of this indirection.